src/android/IntegrityNotificationReceiver.java

```diff
@@ -15,11 +15,11 @@
 
     public IntegrityNotificationReceiver() {
         // The automatically created receiver needs a default constructor
-        Log.d(this, TAG, "noarg constructor called");    
+        Log.d(TAG, "noarg constructor called");    
     }
 
     public IntegrityNotificationReceiver(Context context) {
-        Log.d(this, TAG, "constructor called with arg " + context);
+        Log.d(TAG, "constructor called with arg " + context);
```

Comment:
Are you sure this works? I believe the emission log library requires the context because it writes to the database. The standard android log does not.
https://github.com/e-mission/e-mission-phone/wiki/Troubleshooting-tips-(FAQ)#logging-to-the-database

You can't use the e-mission log for the first constructor since it does not have a context, but all the other methods have a context passed in, and you can use them. See the other receivers for examples.